fiftyone/core/document.py

```diff
@@ -847,9 +847,13 @@ def _reload_parents(self):
 
 
 def _merge_lists(dst, src, overwrite=True):
-    if src is None:
-        return
-
+    if overwrite:
+        # in cases where __eq__ has been defined such that two
+        # instances without the exact same property values returns as
+        # equal, we want to make sure we persist src properties over dst
+        dst, src = src, dst
+    if not dst:
+        dst = []
```

Comment:
You can't create a new array here because this function processes `dst` in place. It doesn't return a new reference